### PR TITLE
fix (separator): prevent bottom separator overlapping content

### LINKED ISSUE #3061: Bottom Separator not going to the back of content
**Describe the bug**
**_Note: This issue is from the affiliate who encountered bugs when testing out Stackable_**
- Related to: https://www.notion.so/39817c6c8fea42e8ab38e2ae6dae36e4
- Applies to all blocks with Bottom Separator
    - [ ]  Columns/Container
    - [ ]  Feature
    - [ ]  Feature Grid
    - [ ]  Hero
    - [ ]  Call to Action
- Issue is visible if Bottom Separator is enabled (make sure Top Separator is disabled)

**To Reproduce**
1. Add a new page
2. Add any blocks mentioned above (e.g. Columns)
3. (optional) Add inner blocks if none yet
4. In Inspector, enable Bottom Separator
5. Make sure to add Height to see issue
6. See that the bottom separator overlaps the inner blocks even if "Bring in Front" is disabled

**Expected behavior**
- Bottom Separator should not overlap inner blocks if "Bring in Front" is disabled

**Workaround**
- Add this code to Custom CSS:
```
.stk-block-columns .stk-separator {
z-index: -1;
}
```

**Screenshots**

https://github.com/gambitph/Stackable/assets/103395655/83ba1a0e-bea1-48a6-b9c9-a0ed55c53a9c

### PR DESCRIPTION
**Pull request by prconcepcion**
fixes #3061

### CODE REVIEW

File: src/block-components/separator/style.scss [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 .stk-block {
 	&:is(.stk-has-top-separator, .stk-has-bottom-separator) {
-		> *:not(.stk-video-background) { // Don't change the order of any video backgrounds, they stay at the back.
+		> *:not(.stk-video-background, .stk-separator) { // Don't change the order of any video backgrounds, they stay at the back.
 			z-index: 2;
+			position: relative;
```
Comment (bfintal) [inline src/block-components/separator/style.scss:11]: can we not place a `position: relative` here because this might cause unwanted side effects.

**Review by bfintal — CHANGES_REQUESTED** [review]

File: src/block-components/separator/style.scss [hunk 2]
```diff
@@ -2,8 +2,9 @@
...
 		}
 
 		.stk-separator ~ *:not(.stk-separator, .stk-video-background) {
```
Comment (prconcepcion) [inline src/block-components/separator/style.scss:14]: @bfintal when I only add a top separator, this rule applies that's why I thought it was fine to add that

File: src/block-components/separator/style.scss [hunk 3]
```diff
@@ -2,16 +2,26 @@
 
 .stk-block {
 	&:is(.stk-has-top-separator, .stk-has-bottom-separator) {
-		> *:not(.stk-video-background) { // Don't change the order of any video backgrounds, they stay at the back.
+		> *:not(.stk-video-background, .stk-separator) { // Don't change the order of any video backgrounds, they stay at the back.
 			z-index: 2;
 		}
 
+		:is(.stk-block-hero__content, .stk-block-call-to-action__content) {
```
Comment (bfintal) [inline src/block-components/separator/style.scss:10]:
```suggestion
		// This is for the CTA and Hero block
		:is(.stk-block-hero__content, .stk-block-call-to-action__content) {
```

File: src/block-components/separator/style.scss [hunk 4]
```diff
@@ -2,16 +2,27 @@
...
+			position: relative;
+		}
+
 		.stk-separator ~ *:not(.stk-separator, .stk-video-background) {
 			z-index: 5;
 			position: relative;
 		}
 	}
 
+	&.stk-block-feature:is(.stk-has-bottom-separator) {
```
Comment (bfintal) [inline src/block-components/separator/style.scss:21]:
```suggestion
	// This is for the Feature block
	&.stk-block-feature:is(.stk-has-bottom-separator) {
```

**Comment by bfintal** [comment]
@prconcepcion Looks like this PR has issues with regards to background tints / gradients going above the separator. The changes from this PR have been reversed from the `develop` branch. Please create a new PR with the adjusted fix

**Pull request merged**